statsmodels/stats/base.py

```diff
@@ -99,3 +102,97 @@ def summary(self):
         text += '\n'.join(('%s  %6.4g' % (pairs, pv) for (pairs, pv) in
                 zip(self.all_pairs_names, self.pval_corrected())))
         return text
+
+
+class Hypothesis(object):
+
+    def __init__(self, null, alternative):
+        self._null = null
+        self._alternative = alternative
+
+    @property
+    def null(self):
+        return self._null
+
+    @property
+    def alternative(self):
+        return self._alternative
+
+    def __str__(self):
+        return ("Hypotheses:\n\t* H0: {0}\n\t* H1: {1}"
+                .format(self._null, self._alternative))
+
+
+class CriticalValues(object):
+
+    def __init__(self, crit_dict):
+        self._crit_dict = crit_dict
+
+    @property
+    def crit_dict(self):
+        return self._crit_dict
+
+    def __str__(self):
+        items = sorted(self._crit_dict.items(),
+                       key=lambda item: int(item[0].strip("%")))
+
+        critical_values = map(lambda item: "[{0}] = {1}".format(*item),
+                              items)
+
+        return "Critical values:\n" + ", ".join(critical_values)
+
+
+class Statistics(object):
+
+    def __init__(self, **kwargs):
+        self.__dict__.update(kwargs)
+
+    def __str__(self):
+        items = map(lambda item: "{0} = {1}".format(*item),
+                    self.__dict__.items())
+
+        return "Statistics:\n" + ", ".join(items)
+
+
+@nottest
+class TestResult(object):
+
+    _options = [
+        "test_name",
+        "hypothesis",
+        "statistics",
+        "critical_values"
+    ]
+
+    def __init__(self, test_name, statistics, **kwargs):
+        self.test_name = test_name
+        self.statistics = statistics
+
+        for key, value in kwargs.items():
+            if key in TestResult._options:
+                setattr(self, key, value)
+
+    def __getattr__(self, item):
```

Comment:
My thinking here is to preserve interface compatibility between old-style test results (e.g. `Bunch`, or `ResStore`) and new-style tests where possible. I intend `Statistics` to be a new-style store object, against which the `TestResult` will always check if the attribute cannot be found on itself.

If it is found in the statistics attribute, the item there is returned and a `DepreciationWarning` is raised explaining the new behaviour. This way we can start using the new-style test results without breaking too much of existing, external code.